### PR TITLE
[AIRFLOW-6568] Add some more entries (Emacs related files) to .gitignore 

### PR DESCRIPTION
**Pull request by sarutak**
Emacs generates some types of backup files.
They should be ignored by the repository.

---
Issue link: [AIRFLOW-6568](https://issues.apache.org/jira/browse/AIRFLOW-6568)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -152,6 +157,7 @@ node_modules
 npm-debug.log*
 derby.log
 metastore_db
+yarn.lock
```
Comment (kaxil) [inline .gitignore]: We do store yarn.lock file: https://github.com/apache/airflow/blob/master/airflow/www/yarn.lock

File: setup.cfg [hunk 2]
```diff
@@ -15,6 +15,9 @@
 # specific language governing permissions and limitations
 # under the License.
 
+
+
+
```
Comment (kaxil) [inline setup.cfg]: Unrelated change?

Comment (sarutak) [reply setup.cfg]: Sorry it might be accidentally changed. I'll revert this.

File: .gitignore (hunk 1 above)
Comment (sarutak) [reply .gitignore]: Yes I notice this one.
What I'd like to ignore is one in the root of the source tree.
But I noticed, the yarn.lock in the root is mistakenly generated ( Maybe, I have run `yarn` command at the root...),
Anyway, I revert this piece of change and modify the description.
Thanks.

File: .gitignore [hunk 3]
```diff
@@ -129,6 +129,11 @@ ENV/
 # vim
 *.swp
 
+# Emacs
+*.~
+*#*#
+*.#*
```
Comment (kaxil) [inline .gitignore]: Does the following list cover all the needed files for you?

```
*~
\#*\#
/.emacs.desktop
/.emacs.desktop.lock
*.elc
auto-save-list
tramp
.\#*
```

Those are from https://github.com/github/gitignore/blob/master/Global/Emacs.gitignore

Comment (sarutak) [reply .gitignore]: I usually need those three entries I added but it's more comprehensive. Thanks!

File: .gitignore [hunk 4]
```diff
@@ -129,6 +129,16 @@ ENV/
 # vim
 *.swp
 
+# Emacs
+*~
+\#*\#
+/.emacs.desktop
```
Comment (ashb) [inline .gitignore:135]: You can (and should) configure your git with a global git exclude list so you don't need every project to support your choice of editor. https://git-scm.com/docs/gitignore

> 

Patterns which a user wants Git to ignore in all situations (e.g., backup or temporary files generated by the user’s editor of choice) generally go into a file specified by core.excludesFile in the user’s ~/.gitconfig. Its default value is $XDG_CONFIG_HOME/git/ignore. If $XDG_CONFIG_HOME is either not set or empty, $HOME/.config/git/ignore is used instead.

**Comment by ashb** [comment]
Please update the PR title/commit message

File: .gitignore (hunk 4 above)
Comment (sarutak) [reply .gitignore:135]: OK, it's reasonable. So why `.swp` is included?

Comment (ashb) [reply .gitignore:135]: Cos people don't know about this. I'm okay accepting this PR, just the more people I let know the easier their life is in the long run :)

Comment (sarutak) [reply .gitignore:135]: Yeah, I didn't know about that and got a better way. Thanks!

Comment (potiuk) [reply .gitignore:135]: And what does not help is most of the scaffolding where you generate .gitignore files already contains most of the "common" exclusions so adding them elsewhere seems inconsistent even if it avoids repetition.

**Review by potiuk — APPROVED** [review]
